Assets/Scripts/Terrain/TerrainTexturing.cs

```diff
@@ -219,7 +222,7 @@ public void Execute(int index)
                 }
                 // Beach texture
                 // Adds a little +/- randomness to threshold so beach line isn't too regular
-                if (height <= beachElevation + (JobRand.Next(-15000000, 15000000) / 10000000f))
+                if (height <= beachElevation + Unity.Mathematics.Random.CreateFromIndex((uint)index).NextFloat(-1.5f, 1.5f))
```

Comment:
Hi, could you explain more about this change please? The JobRand helper class providing basic thread safe access to system random was the best solution I could find when migrating the original code into Jobs.

Why are you changing it here? Is the helper class not compatible with burst?

Reply:
> Why are you changing it here? Is the helper class not compatible with burst?
Yes exactly, `System.Random`, being a managed type, is not compatible with Burst

Reply:
Before ending with this particular solution I dig into this and run some tests to figure out how to generate random numbers without memory allocation with `Unity.Mathematics.Random`. You can look through these numerical results in my post [here](https://gist.github.com/andrew-raphael-lukasik/b767c82969eb67ae76b6351bd31bb558?permalink_comment_id=4233622#gistcomment-4233622)